packages/nocodb/src/middlewares/extract-ids/extract-ids.middleware.ts

```diff
@@ -255,39 +257,41 @@ export class AclMiddleware implements NestInterceptor {
     if (req?.user?.is_api_token && blockApiTokenAccess) {
       NcError.forbidden('Not allowed with API token');
     }
+
+    const allRoles = await Noco.ncMeta
+      .knex(MetaTable.ROLES)
+      .select('name', 'label');
+
     if (
       (!allowedRoles || allowedRoles.some((role) => roles?.[role])) &&
-      !(
-        roles?.creator ||
-        roles?.owner ||
-        roles?.editor ||
-        roles?.viewer ||
-        roles?.commenter ||
-        roles?.[OrgUserRoles.SUPER_ADMIN] ||
-        roles?.[OrgUserRoles.CREATOR] ||
-        roles?.[OrgUserRoles.VIEWER]
-      )
+      !allRoles.some((item) => Object.keys(roles).includes(item.name))
```

Comment:
This condition is for cases when a user doesn't have permission because we throw an exception in this block. We allow the user to do something when the condition in this block doesn't succeed.